core/net/rpl/rpl-dag.c

```diff
@@ -691,29 +696,31 @@ rpl_find_parent_any_dag(rpl_instance_t *instance, uip_ipaddr_t *addr)
 }
 /*---------------------------------------------------------------------------*/
 rpl_dag_t *
-rpl_select_dag(rpl_instance_t *instance, rpl_parent_t *p)
+rpl_select_dag(rpl_instance_t *instance, rpl_dag_t *dag)
 {
+  /* This function (1) updates 'dag' (2) chooses a new current DAG in 'instance'
+   * and (3) updates the current DAG. */
   rpl_parent_t *last_parent;
-  rpl_dag_t *dag, *end, *best_dag;
+  rpl_dag_t *d, *end, *best_dag;
```

Comment:
Is there a bit more informative name that we could use for d? Single character variables (beyond i,j) is not optimal when it comes to readability.